app/src/gui/fields/TakePhoto.tsx

```diff
@@ -262,100 +257,90 @@ export class TakePhoto extends React.Component<
-      error_text = (
-        <span {...this.props['ErrorTextProps']}>{error as string}</span>
-      );
+      props.form.setFieldError(props.field.name, err.message);
     }
+  };
 
-    // https://mui.com/components/image-list/#masonry-image-list
-    // Masonry image lists use dynamically sized container heights
-    // that reflect the aspect ratio of each image. This image list
-    // is best used for browsing uncropped peer content.
-    // But it doesn't look like we support masonry right now.
-    //
-    // It also looks like we don't have multiple photos being returned...
+  const error = props.form.errors[props.field.name];
+  const errorText = error ? (
+    <span {...props.ErrorTextProps}>{error as string}</span>
+  ) : (
+    <span {...props.NoErrorTextProps}></span>
+  );
 
-    // We have two properties available here - both are optional
-    const title = this.props.label;
-    const helperText =
-      this.props.helpertext ?? this.props.helperText ?? undefined;
+  const title = props.label;
+  const helperText = props.helpertext ?? props.helperText ?? undefined;
 
-    return (
-      <MediaQueryWrapper>
-        {isMobile => (
-          <div>
-            {title && <h3>{title}</h3>}
-            {helperText && <p>{helperText}</p>}
-            <Button
-              variant="contained"
-              color="primary"
-              fullWidth={isMobile ? true : false}
-              onClick={async () => {
-                await this.takePhoto();
-              }}
-            >
-              Take photo
-              <span style={{width: 10}} />
-              <CameraAltIcon />
-            </Button>
-            <FAIMSImageList
-              images={this.state.images}
-              setopen={(path: string) =>
-                this.setState({open: true, photopath: path})
-              }
-              setimage={(newfiles: Array<any>) => {
-                this.props.form.setFieldValue(
-                  this.props.field.name,
-                  newfiles,
-                  true
-                );
-              }}
-              disabled={this.props.disabled ?? false}
-              fieldName={this.props.field.name}
-            />
-            <Typography variant="caption" color="textSecondary">
-              {error_text}{' '}
-            </Typography>
-            <FaimsAttachmentManagerDialog
-              project_id={this.props.form.values['_project_id']}
-              open={this.state.open}
-              setopen={() => this.setState({open: false})}
-              filedId={this.props.id}
-              path={this.state.photopath}
-              isSyncing={this.props.issyncing}
-            />
-          </div>
-        )}
-      </MediaQueryWrapper>
-    );
-  }
-}
+  return (
+    <div>
+      {title && <h3>{title}</h3>}
+      {helperText && <p>{helperText}</p>}
+      <Button
+        variant="contained"
+        color="primary"
+        fullWidth={isMobile ? true : false}
+        onClick={takePhoto}
```

Comment:
This change looks similar to the one that broke it before which you fixed - are you sure just takePhoto is okay, you don't need async () => {await takePhoto()}; ?

Reply:
That was because it was an object based component and it forgot what `this` was because of the call, here we're just calling a fn which doesn't have this so no bug.